test_util/integration_test.py

```diff
@@ -1290,75 +1292,105 @@ def wait_for_endpoint():
 
     r = requests.get(test_cache_url)
 
+    # Handy for diagnosing strange signal service test behavior, not sure if we should
+    # leave these in for master branch, or remove for the final PR.
+    print('TESTING SIGNAL RETURN:\n{}'.format(r.text))
+    print('CACHE SERVER STATUS:\n{}'.format(r.status_code))
+
     r_data = json.loads(r.json())
 
     cluster.destroy_marathon_app(signal_app_definition['id'])
     cluster.destroy_marathon_app(test_server_app_definition['id'])
 
     exp_data = {
-            'Event': 'health',
-            'UserId': '',
-            'ClusterId': '',
-            'Properties': {
-                'provider': 'onprem',
-                'source': 'cluster',
-                'clusterId': '',
-                'customerKey': '',
-                'environmentVersion': '',
-                'variant': 'open'}
-            }
+        'diagnostics': {
+            'event': 'health',
+            'anonymousId': 'open /var/lib/dcos/cluster-id: no such file or directory',
+            'properties': {}
+        },
+        'cosmos': {
+            'event': 'package_list',
+            'anonymousId': 'open /var/lib/dcos/cluster-id: no such file or directory',
+            'properties': {}
+        },
+        'mesos': {
+            'event': 'mesos_track',
+            'anonymousId': 'open /var/lib/dcos/cluster-id: no such file or directory',
+            'properties': {}
+        }
+    }
+
+    # Generic properties which are the same between all tracks
+    generic_properties = {
+        'provider': 'onprem',
+        'source': 'cluster',
+        'clusterId': 'open /var/lib/dcos/cluster-id: no such file or directory',
+        'customerKey': '',
+        'environmentVersion': '',
+        'variant': 'open'
+    }
+
+    # Insert the generic property data which is the same between all signal tracks
+    for k, v in generic_properties.items():
+        exp_data['diagnostics']['properties'][k] = v
+        exp_data['cosmos']['properties'][k] = v
+        exp_data['mesos']['properties'][k] = v
 
+    # Insert all the diagnostics data programmatically
     master_units = [
```

Comment:
Please don't change this indentation here. Makes a lot of noise in the diff (Can change whitespace, just do it separately from when you're making semantic changes).